dbms/src/DataStreams/AddingDefaultsBlockInputStream.cpp

```diff
@@ -0,0 +1,207 @@
+#include <Common/typeid_cast.h>
+#include <Functions/FunctionHelpers.h>
+#include <Interpreters/ExpressionActions.h>
+#include <Interpreters/evaluateMissingDefaults.h>
+#include <DataStreams/AddingDefaultsBlockInputStream.h>
+
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnsCommon.h>
+#include <Columns/ColumnDecimal.h>
+#include <Columns/ColumnConst.h>
+#include <Columns/FilterDescription.h>
+
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypesDecimal.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <DataTypes/DataTypeUUID.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeFixedString.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int SIZES_OF_COLUMNS_DOESNT_MATCH;
+    extern const int TYPE_MISMATCH;
+}
+
+
+AddingDefaultsBlockInputStream::AddingDefaultsBlockInputStream(const BlockInputStreamPtr & input,
+                                                               const ColumnDefaults & column_defaults_,
+                                                               const Context & context_)
+    : column_defaults(column_defaults_),
+      context(context_)
+{
+    children.push_back(input);
+    header = input->getHeader();
+}
+
+
+Block AddingDefaultsBlockInputStream::readImpl()
+{
+    Block res = children.back()->read();
+    if (!res)
+        return res;
+
+    if (column_defaults.empty())
+        return res;
+
+    const BlockMissingValues & delayed_defaults = children.back()->getMissingValues();
+    if (delayed_defaults.empty())
+        return res;
+
+    Block evaluate_block{res};
+    for (const auto & column : column_defaults)
+    {
+        /// column_defaults contain aliases that could be ommited in evaluate_block
```

Comment:
Comment is a bit misleading - the code doesn't execute specifically for ALIAS columns. I guess the reason is that we remove these columns so that we can calculate default values for them.